samples/sample_common/src/vaapi_utils_drm.cpp

```diff
@@ -22,169 +22,104 @@ or https://software.intel.com/en-us/media-client-solutions-support.
-            if (fgets(str, sizeof(str), file))
-            {
-                class_id = strtol(str, NULL, 16);
-            }
-            fclose(file);
-
-            if (MFX_PCI_DISPLAY_CONTROLLER_CLASS == (class_id >> 16))
-            {
-                // obtaining device vendor id
-                snprintf(file_name, sizeof(file_name)/sizeof(file_name[0]), "%s/%s/%s", MFX_PCI_DIR, dir_entries[i]->d_name, "vendor");
-                file = fopen(file_name, "r");
-                if (file)
-                {
-                    if (fgets(str, sizeof(str), file))
-                    {
-                        vendor_id = strtol(str, NULL, 16);
-                    }
-                    fclose(file);
-                }
-                // obtaining device id
-                snprintf(file_name, sizeof(file_name)/sizeof(file_name[0]), "%s/%s/%s", MFX_PCI_DIR, dir_entries[i]->d_name, "device");
-                file = fopen(file_name, "r");
-                if (file)
-                {
-                    if (fgets(str, sizeof(str), file))
-                    {
-                        device_id = strtol(str, NULL, 16);
-                    }
-                    fclose(file);
-                }
-                // adding valid adapter to the list
-                if (vendor_id && device_id)
-                {
-                    struct mfx_disp_adapters* tmp_adapters = NULL;
-
-                    tmp_adapters = (mfx_disp_adapters*)realloc(adapters,
-                                                               (adapters_num+1)*sizeof(struct mfx_disp_adapters));
-
-                    if (tmp_adapters)
-                    {
-                        adapters = tmp_adapters;
-                        adapters[adapters_num].vendor_id = vendor_id;
-                        adapters[adapters_num].device_id = device_id;
-
-                        ++adapters_num;
-                    }
-                }
-            }
+            adapters.push_back(i);
         }
-        free(dir_entries[i]);
+        close(fd);
     }
-    if (entries_num) free(dir_entries);
-    if (p_adapters) *p_adapters = adapters;
-
-    return adapters_num;
 }
 
 DRMLibVA::DRMLibVA(int type)
     : CLibVA(type)
     , m_fd(-1)
 {
-    const mfxU32 IntelVendorID = 0x8086;
-    //the first Intel adapter is only required now, the second - in the future
+    // The first Intel adapter is only required now, the second - in the future
     const mfxU32 numberOfRequiredIntelAdapter = 1;
-    const char nodesNames[][8] = {"renderD", "card"};
 
     VAStatus va_res = VA_STATUS_SUCCESS;
     mfxStatus sts = MFX_ERR_NONE;
     int major_version = 0, minor_version = 0;
 
-    mfx_disp_adapters* adapters = NULL;
-    int adapters_num = mfx_init_adapters(&adapters);
+    std::vector<mfxU32> adapters;
+    get_intel_adapters(adapters);
+    
+    if (adapters.size() != numberOfRequiredIntelAdapter)
+        throw std::range_error("Number of required Intel adapters was not found");
```

Comment:
Message is ugly. Please, change. "Intel GPU was not found" would be better, I think.